kong/router.lua

```diff
@@ -5,8 +5,13 @@ local bit           = require "bit"
 
 
 local hostname_type = utils.hostname_type
+local subsystem     = ngx.config.subsystem
+local get_method    = ngx.req.get_method
```

Comment:
One concern I have about this is that it is easy to add another localized `ngx.` function here and forget to add the equivalent code in `_set_ngx` below (and end up with a "partially mocked" `ngx`). If we are going the path of `_set_ngx`, I would rather declare these variables with no value, and then call `_set_ngx(ngx)` at the top-level of the module, to make it more evident that the initialization of those variables is special.

Reply:
partially mocking well may be the intent of the user?